src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -3694,54 +3705,58 @@ string WriteProperty(byte[] propertyName, byte[] value)
             Assert.Equal(expected3, result);
         }
 
+        private const string InvalidUtf16 = "\"\\uFFFDa\"";
+        private const string ValidUtf16 = "\"\\uD801\\uDC37\"";
+
         [Theory]
-        [InlineData(true, true)]
-        [InlineData(true, false)]
-        [InlineData(false, true)]
-        [InlineData(false, false)]
-        public void InvalidUTF16(bool formatted, bool skipValidation)
+        [InlineData(true,
+            "{" + InvalidUtf16 + ":" + InvalidUtf16 + "}",
+            "{" + InvalidUtf16 + ":" + ValidUtf16 + "}",
+            "{" + ValidUtf16 + ":" + InvalidUtf16 + "}",
+            "{" + ValidUtf16 + ":" + ValidUtf16 + "}")]
+        [InlineData(false,
+            "{" + InvalidUtf16 + ":" + InvalidUtf16 + "}",
+            "{" + InvalidUtf16 + ":" + ValidUtf16 + "}",
+            "{" + ValidUtf16 + ":" + InvalidUtf16 + "}",
+            "{" + ValidUtf16 + ":" + ValidUtf16 + "}")]
+        public void UTF16ReplacementCharacters(bool skipValidation, string expected0, string expected1, string expected2, string expected3)
```

Comment:
nit: This seems like an odd test name.
Maybe `InvalidUTF16SurrogatePairsGetReplaced`

Reply:
Also, consider keeping the formatted/skipvalidation permutations.

Reply:
I did not keep `formatted` permutations -- there's another test for that (HighSurrogateMissingGetsReplaced)